projects/packages/forms/src/modules/file-field/view.js

```diff
@@ -0,0 +1,208 @@
+	const i = Math.floor( Math.log( size ) / Math.log( k ) );
+	const formattedSize = parseFloat( ( size / Math.pow( k, i ) ).toFixed( dm ) );
+	const numberFormat = new Intl.NumberFormat( locale, {
+		minimumFractionDigits: dm,
+		maximumFractionDigits: dm,
+	} );
+	return `${ numberFormat.format( formattedSize ) } ${ sizes[ i ] }`;
+};
+
+const { state, callbacks } = store( 'jpDropZone', {
+	state: {},
+	actions: {
+		/**
+		 * Open the file picker dialog.
+		 *
+		 * @param {Event} event - The event object.
+		 */
+		openFilePicker: event => {
+			const fileInput = event.target.parentNode.querySelector( '.jetpack-form-file-field' );
+			if ( fileInput ) {
+				fileInput.click();
+			}
+		},
+
+		/**
+		 * Handle file added event.
+		 *
+		 * @param {Event} event - The event object.
+		 */
+		fileAdded: event => {
+			const files = Array.from( event.target.files );
+			files.forEach( callbacks.addFileToContext );
+		},
+
+		/**
+		 * Handle file dropped event.
+		 *
+		 * @param {DragEvent} event - The drag event object.
+		 */
+		fileDropped: event => {
+			event.preventDefault();
+			if ( event.dataTransfer ) {
+				for ( const item of Array.from( event.dataTransfer.items ) ) {
+					if ( item.webkitGetAsEntry()?.isDirectory ) {
+						return;
+					}
+					callbacks.addFileToContext( item.getAsFile() );
+				}
+			}
+			const context = getContext();
+			context.isDropping = false;
+		},
+
+		/**
+		 * Handle drag over event.
+		 *
+		 * @param {DragEvent} event - The drag event object.
+		 */
+		dragOver: event => {
+			const context = getContext();
+			context.isDropping = true;
+			event.preventDefault();
+		},
+
+		/**
+		 * Handle drag leave event.
+		 */
+		dragLeave: () => {
+			const context = getContext();
+			context.isDropping = false;
+		},
+
+		/**
+		 * Remove a file from the context.
+		 *
+		 * @param {Event} event - The event object.
+		 */
+		removeFile: event => {
+			const context = getContext();
+			const fileId = event.target.dataset.id;
+			context.files = context.files.filter( file => file.id !== fileId );
+			context.hasFiles = context.files.length > 0;
+		},
+	},
+
+	callbacks: {
+		/**
+		 * Add the file to the context.
+		 *
+		 * @param {File} file - The file to add.
+		 */
+		addFileToContext: file => {
+			const reader = new FileReader();
+			reader.readAsDataURL( file );
+			reader.onload = withScope( () => {
+				const context = getContext();
+				const fileId = performance.now() + '-' + Math.random();
+				context.files.push( {
+					name: file.name,
+					url: 'url(' + reader.result + ')',
+					formattedSize: formatBytes( file.size, 2, state.i18n.locale ),
+					hasToken: false,
+					id: fileId,
+				} );
+				context.hasFiles = true;
+				callbacks.uploadFile( file, fileId );
+			} );
+		},
+
+		/**
+		 * Make the endpoint request.
+		 *
+		 * @param {File}   file   - The file to upload.
+		 * @param {string} fileId - The file ID.
+		 */
+		uploadFile: ( file, fileId ) => {
```

Comment:
Same here and in all the other callbacks.

Reply:
Updated this https://github.com/Automattic/jetpack/pull/42161#discussion_r1981828970